x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas.ts

```diff
@@ -110,6 +135,7 @@ export const createRulesSchema = Joi.object({
   tags: tags.default([]),
   to: to.required(),
   type: type.required(),
+  threats,
```

Comment:
I would just default this on the way in as an empty array if it's not set like so:

```ts
threats: threats.default([])
```

Will make the typing and code easier. It's at least what I have done for `references` below and `tags` above when the user doesn't push anything in, I just default them to `[]`

Reply:
Looks like I can default to an empty array and not throw any errors 👍 Will update this too.